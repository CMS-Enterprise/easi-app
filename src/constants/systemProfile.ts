// Loosely grouped by System Subpages

// Team

export const teamSectionKeys = [
  'businessOwners',
  'projectLeads',
  'additional'
] as const;

export const TEAM_SECTION_MEMBER_COUNT_CAP = 5;
export const TOOLS_AND_SOFTWARE_PRODUCT_COUNT_CAP = 5;
export const BUDGET_ITEMS_COUNT_CAP = 5;

// ATO

export const ATO_STATUS_DUE_SOON_DAYS = 90;

// Threat levels match values from cedarThreat.weaknessRiskLevel
export const threatLevelGrades = [
  'Critical',
  'High',
  'Moderate',
  'Low',
  'Not Rated'
] as const;

export const securityFindingKeys = ['total', ...threatLevelGrades] as const;
