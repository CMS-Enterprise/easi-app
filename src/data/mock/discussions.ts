import {
  SystemIntakeGRBReviewDiscussionFragment,
  SystemIntakeGRBReviewerRole,
  SystemIntakeGRBReviewerVotingRole
} from 'gql/gen/graphql';

const mockDiscussions = (
  systemIntakeID: string = '0d191c27-4de3-46d8-9ded-c2e9c6b3fbd6'
): SystemIntakeGRBReviewDiscussionFragment[] => [
  {
    __typename: 'SystemIntakeGRBReviewDiscussion',
    initialPost: {
      __typename: 'SystemIntakeGRBReviewDiscussionPost',
      id: '882357e4-c0b0-44ef-b749-f71879ad7878',
      content:
        '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>',
      votingRole: SystemIntakeGRBReviewerVotingRole.VOTING,
      grbRole: SystemIntakeGRBReviewerRole.SUBJECT_MATTER_EXPERT,
      createdByUserAccount: {
        __typename: 'UserAccount',
        id: '034fa2b3-00ff-4ec6-857e-75291a59df74',
        commonName: 'Adeline Aarons'
      },
      systemIntakeID,
      createdAt: '2024-11-12T21:59:07.368862Z'
    },
    replies: [
      {
        __typename: 'SystemIntakeGRBReviewDiscussionPost',
        id: '4099dbb7-2752-4bf9-a3e1-da225ceb9fae',
        content:
          '<p>Nisi nobis consectetur voluptatem neque tempore. Ea nihil sed beatae?</p>',
        votingRole: SystemIntakeGRBReviewerVotingRole.NON_VOTING,
        grbRole: SystemIntakeGRBReviewerRole.CMCS_REP,
        createdByUserAccount: {
          __typename: 'UserAccount',
          id: '909a7888-4d6f-4bbf-9b9d-71ec1e2c3068',
          commonName: 'Jerry Seinfeld'
        },
        systemIntakeID,
        createdAt: '2024-11-13T21:59:07.368862Z'
      },
      {
        __typename: 'SystemIntakeGRBReviewDiscussionPost',
        id: '47b0081d-de33-4514-b68f-a7e2bbf2610f',
        content:
          '<p>Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?</p>',
        votingRole: SystemIntakeGRBReviewerVotingRole.VOTING,
        grbRole: SystemIntakeGRBReviewerRole.CO_CHAIR_CIO,
        createdByUserAccount: {
          __typename: 'UserAccount',
          id: '601d52be-7baa-4b45-91cd-88b4a5935c3f',
          commonName: 'Cosmo Kramer'
        },
        systemIntakeID,
        createdAt: '2024-11-13T21:59:07.368862Z'
      }
    ]
  }
];

export default mockDiscussions;
