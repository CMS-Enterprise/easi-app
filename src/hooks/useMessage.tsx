import React, {
  createContext,
  ReactNode,
  useContext,
  useEffect,
  useState
} from 'react';
import { useLocation } from 'react-router-dom';

import Alert, { AlertProps } from 'components/shared/Alert';

type MessageProps = {
  message?: string | React.ReactElement;
  alertProps?: Omit<AlertProps, 'children'>;
};

type SetMessage = (
  message?: string | React.ReactElement,
  alertProps?: Omit<AlertProps, 'children'>
) => void;

const MessageContext = createContext<
  | {
      Message: ({ className }: { className?: string }) => JSX.Element | null;
      showMessage: SetMessage;
      showMessageOnNextPage: SetMessage;
    }
  | undefined
>(undefined);

// MessageProvider manages the state necessary for child components to
// use the useMessage hook.
const MessageProvider = ({ children }: { children: ReactNode }) => {
  const [queuedMessage, setQueuedMessage] = useState<MessageProps>({});
  const [message, setMessage] = useState<MessageProps>({});

  const location = useLocation();

  const [lastPathname, setLastPathname] = useState(location.pathname);

  const Message = (props: { className?: string }) => {
    // If no message, return null
    if (!message.message) return null;

    // If message has alert props, wrap in component
    if (message.alertProps) {
      return (
        <Alert {...message.alertProps} {...props}>
          {message.message}
        </Alert>
      );
    }

    // Return plain message
    return <>{message.message}</>;
  };

  useEffect(() => {
    if (lastPathname !== location.pathname) {
      setMessage(queuedMessage);
      setQueuedMessage({});
      setLastPathname(location.pathname);
    }
  }, [message, queuedMessage, lastPathname, location.pathname]);

  return (
    <MessageContext.Provider
      value={{
        Message,
        showMessage: (value, alertProps) =>
          setMessage({
            message: value,
            alertProps
          }),
        showMessageOnNextPage: (value, alertProps) =>
          setQueuedMessage({
            message: value,
            alertProps
          })
      }}
    >
      {children}
    </MessageContext.Provider>
  );
};

// useMessage is a hook that provides the ability to show notifications to the
// user. Components will use the hook to get access to two methods:
//
// showMessage: display a message to the user immediately. This is useful
// when an interaction won't cause an immediate redirect to a new page.
//
// showMessageOnNextPage: schedule a message to be shown on the next
// visited by the user. Pages are identified using their URL and therefore
// messages won't appear until the URL changes.
//
// Currently there is only support for showing a single message.
const useMessage = () => {
  const context = useContext(MessageContext);
  if (context === undefined) {
    throw new Error('useMessage must be used within a MessageProvider');
  }

  return context;
};

export default useMessage;
export { useMessage, MessageProvider };
