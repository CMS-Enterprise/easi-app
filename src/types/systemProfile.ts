import {
  securityFindingKeys,
  teamSectionKeys,
  threatLevelGrades
} from 'constants/systemProfile';
import {
  GetSystemProfile,
  /* eslint-disable camelcase */
  GetSystemProfile_cedarAuthorityToOperate,
  GetSystemProfile_cedarSystemDetails_cedarSystem,
  GetSystemProfile_cedarSystemDetails_deployments_dataCenter,
  GetSystemProfile_cedarSystemDetails_roles,
  GetSystemProfile_cedarSystemDetails_urls
  /* eslint-enable camelcase */
} from 'queries/types/GetSystemProfile';
import {
  tempATOProp,
  tempBudgetProp,
  tempProductsProp,
  tempSubSystemProp,
  tempSystemDataProp
} from 'views/Sandbox/mockSystemData';

import { CedarAssigneeType } from './graphql-global-types';

// Loosely grouped by System Subpages

// Team

export type TeamSectionKey = typeof teamSectionKeys[number];

export interface CedarRoleAssigneePerson
  // eslint-disable-next-line camelcase
  extends GetSystemProfile_cedarSystemDetails_roles {
  assigneeType: CedarAssigneeType.PERSON;
}

export interface UsernameWithRoles {
  assigneeUsername: string;
  roles: CedarRoleAssigneePerson[];
}

// ATO

export type AtoStatus =
  | 'Active'
  | 'Due Soon'
  | 'In Progress'
  | 'Expired'
  | 'No ATO';

export type ThreatLevel = typeof threatLevelGrades[number];

export type SecurityFindings = Record<
  typeof securityFindingKeys[number],
  number
>;

// General

export type SubpageKey =
  | 'home'
  | 'details'
  | 'team'
  | 'funding-and-budget'
  | 'tools-and-software'
  | 'ato'
  | 'lifecycle-id'
  | 'section-508'
  | 'sub-systems'
  | 'system-data'
  | 'documents';

/**
 * A set of some Cedar Role Type Ids used in the client.
 * So far it is certain team members and points of contact.
 * This is a borrowed set from `API/role/type/alfabet`.
 */
// eslint-disable-next-line no-shadow
export enum RoleTypeId {
  API_CONTACT = '238-58-0',
  BUSINESS_OWNER = '238-17-0',
  COR = '238-35-0',
  ISSO = '238-29-0',
  PROJECT_LEAD = '238-32-0',
  SYSTEM_BUSINESS_QUESTION_CONTACT = '238-51-0',
  SYSTEM_DATA_CENTER_CONTACT = '238-52-0',
  SYSTEM_ISSUES_CONTACT = '238-50-0',
  SYSTEM_MAINTAINER = '238-28-0'
}

// Development Tags

export type DevelopmentTag = 'Agile Methodology';
// | 'AI Technologies'
// | 'Healthcare Quality'
// | 'Health Insurance Program';

// Urls and Locations

export type UrlLocationTag = 'API endpoint' | 'Versioned code respository';

// eslint-disable-next-line camelcase
export interface UrlLocation extends GetSystemProfile_cedarSystemDetails_urls {
  // eslint-disable-next-line camelcase
  deploymentDataCenterName?: GetSystemProfile_cedarSystemDetails_deployments_dataCenter['name'];
  tags: UrlLocationTag[];
}

export interface SystemProfileData extends GetSystemProfile {
  // The original id type can be null, in which case this object is not created
  id: string;
  // eslint-disable-next-line camelcase
  ato?: GetSystemProfile_cedarAuthorityToOperate;
  atoStatus?: AtoStatus;
  locations?: UrlLocation[];
  // eslint-disable-next-line camelcase
  businessOwners: GetSystemProfile_cedarSystemDetails_roles[];
  developmentTags?: DevelopmentTag[];
  numberOfContractorFte?: number;
  numberOfFederalFte?: number;
  numberOfFte?: number;
  personRoles: CedarRoleAssigneePerson[];
  productionLocation?: UrlLocation;
  // eslint-disable-next-line camelcase
  status: GetSystemProfile_cedarSystemDetails_cedarSystem['status'];
  usernamesWithRoles: UsernameWithRoles[];

  // Remaining mock data stubs
  activities?: tempATOProp[];
  budgets?: tempBudgetProp[];
  products?: tempProductsProp[];
  subSystems?: tempSubSystemProp[];
  systemData?: tempSystemDataProp[];
}

export interface SystemProfileSubviewProps {
  system: SystemProfileData;
}
