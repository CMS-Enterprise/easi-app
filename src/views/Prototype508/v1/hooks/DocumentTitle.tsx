import { useEffect, useRef } from 'react';

function useDocumentTitle(title: string) {
  const defaultTitle = useRef(document.title);

  useEffect(() => {
    document.title = title;
  }, [title]);

  useEffect(() => {
    const newTitle = defaultTitle.current;
    return () => {
      if (newTitle) {
        document.title = newTitle;
      }
    };
  }, []);
}

export default useDocumentTitle;
