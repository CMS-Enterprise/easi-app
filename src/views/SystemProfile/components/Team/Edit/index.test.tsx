import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/react-testing';
import { render, within } from '@testing-library/react';
import i18next from 'i18next';

import { usernamesWithRoles } from 'data/mock/systemProfile';
import { MessageProvider } from 'hooks/useMessage';

import EditTeam from '.';

const cedarSystemId = '000-0000-1';

describe('Edit team page', () => {
  it('Renders the edit team page', async () => {
    const { findByRole, getByRole, getByTestId } = render(
      <MemoryRouter initialEntries={[`/systems/${cedarSystemId}/team/edit`]}>
        <MessageProvider>
          <MockedProvider>
            <Route path="/systems/:systemId/team/edit">
              <EditTeam
                name="Easy Access to System Information"
                team={usernamesWithRoles}
                numberOfFederalFte={6}
                numberOfContractorFte={4}
              />
            </Route>
          </MockedProvider>
        </MessageProvider>
      </MemoryRouter>
    );

    expect(
      await findByRole('heading', {
        name: i18next.t('systemProfile:singleSystem.editTeam.title')
      })
    ).toBeInTheDocument();

    /* Employee fields hidden until work to update in CEDAR is completed */

    // const employeesFieldFederal = getByRole('spinbutton', {
    //   name: i18next.t('systemProfile:singleSystem.editTeam.federalEmployees')
    // });
    // expect(employeesFieldFederal).toHaveValue(6);

    // const employeesFieldContractors = getByRole('spinbutton', {
    //   name: i18next.t('systemProfile:singleSystem.editTeam.contractors')
    // });
    // expect(employeesFieldContractors).toHaveValue(4);

    expect(
      getByRole('button', {
        name: i18next.t('systemProfile:singleSystem.editTeam.addNewTeamMember')
      })
    ).toBeInTheDocument();

    // Check number of team cards rendered
    const { getAllByTestId } = within(getByTestId('teamCardGroup'));
    const teamCards = getAllByTestId('Card');
    expect(teamCards).toHaveLength(usernamesWithRoles.length);

    // Check for edit and remove buttons on team cards
    expect(
      within(teamCards[0]).getByRole('button', { name: 'Edit roles' })
    ).toBeInTheDocument();
    expect(
      within(teamCards[0]).getByRole('button', { name: 'Remove team member' })
    ).toBeInTheDocument();
  });
});
