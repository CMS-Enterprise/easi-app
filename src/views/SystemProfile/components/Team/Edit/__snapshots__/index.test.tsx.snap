// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Workspace team page > renders 1`] = `
<DocumentFragment>
  <div
    class="grid-container margin-bottom-10"
    data-testid="gridContainer"
  >
    <h1
      class="margin-bottom-05"
    >
      Manage system team
    </h1>
    <p
      class="margin-top-0 margin-bottom-1 text-darkest font-body-lg line-height-body-5 text-light"
    >
      for Easy Access to System Information
    </p>
    <p
      class="margin-y-1 font-body-md line-height-body-4 text-light"
    >
      Add or remove team members and manage their roles.
    </p>
    <a
      class="usa-link display-flex flex-align-center margin-top-2 margin-bottom-6 line-height-body-4 text-primary"
      href="/systems/000-0000-1/workspace"
    >
      <svg
        class="usa-icon margin-right-1"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      Return to system workspace
    </a>
    <h4
      class="margin-top-5 margin-bottom-1"
    >
      Team members
    </h4>
    <button
      class="usa-button margin-bottom-05"
      data-testid="button"
      type="button"
    >
      Add another team member
    </button>
    <div>
      <form
        class="usa-form usa-search margin-y-3"
        data-testid="table-client-filter"
        role="search"
      >
        <label
          class="usa-sr-only"
          data-testid="label"
          for="workspace-team-table-search"
        >
          Search table
        </label>
        <input
          class="usa-input"
          data-testid="textInput"
          id="workspace-team-table-search"
          name="Team members Search"
          role="searchbox"
          type="search"
          value=""
        />
        <button
          class="usa-button grid-row flex-justify-center flex-align-center no-pointer"
          data-testid="button"
          type="submit"
        >
          <svg
            class="usa-icon usa-icon--size-3"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </button>
      </form>
      <table
        class="usa-table usa-table--borderless src-components-Table-Table-module__fullwidth--fOyj6"
        data-testid="table"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              aria-sort="none"
              class="border-bottom-2px"
              colspan="1"
              role="columnheader"
              scope="col"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <button
                class="usa-button usa-button--unstyled width-full display-flex"
                data-testid="button"
                style="cursor: pointer;"
                title="Toggle SortBy"
                type="button"
              >
                <div
                  class="flex-fill text-no-wrap"
                >
                  Name
                </div>
                <div
                  class="position-relative width-205 margin-left-05"
                >
                  <svg
                    class="usa-icon margin-left-05 position-absolute"
                    data-testid="caret--sort"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
                    />
                  </svg>
                </div>
              </button>
            </th>
            <th
              aria-sort="descending"
              class="border-bottom-2px"
              colspan="1"
              role="columnheader"
              scope="col"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <button
                class="usa-button usa-button--unstyled width-full display-flex"
                data-testid="button"
                style="cursor: pointer;"
                title="Toggle SortBy"
                type="button"
              >
                <div
                  class="flex-fill text-no-wrap"
                >
                  Role
                </div>
                <div
                  class="position-relative width-205 margin-left-05"
                >
                  <svg
                    class="usa-icon margin-left-05 position-absolute"
                    data-testid="caret--down"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </div>
              </button>
            </th>
            <th
              aria-sort="none"
              class="border-bottom-2px"
              colspan="1"
              role="columnheader"
              scope="col"
            >
              <button
                class="usa-button usa-button--unstyled width-full display-flex"
                data-testid="button"
                type="button"
              >
                <div
                  class="flex-fill text-no-wrap"
                >
                  Actions
                </div>
                <div
                  class="position-relative width-205 margin-left-05"
                >
                  <svg
                    class="usa-icon margin-left-05 position-absolute"
                    data-testid="caret--sort"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
                    />
                  </svg>
                </div>
              </button>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              <div
                class="display-flex flex-align-center"
              >
                <div
                  class="display-flex flex-align-center flex-justify-center minw-4 circle-4 bg-blue-10 "
                  data-testid="avatar--basic"
                >
                  AL
                </div>
                <span
                  class="margin-left-1"
                >
                  Aletha Leannon
                </span>
              </div>
            </td>
            <td
              role="cell"
            >
              Business Owner
            </td>
            <td
              role="cell"
            >
              <div
                class="text-no-wrap"
              >
                <a
                  class="usa-link text-primary"
                  href="/systems/000-0000-1/team/edit/team-member?workspace"
                >
                  Edit roles
                </a>
                <button
                  class="usa-button usa-button--unstyled margin-left-1 text-error"
                  data-testid="button"
                  type="button"
                >
                  Remove
                </button>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              <div
                class="display-flex flex-align-center"
              >
                <div
                  class="display-flex flex-align-center flex-justify-center minw-4 circle-4 bg-orange-warm-10 "
                  data-testid="avatar--basic"
                >
                  EH
                </div>
                <span
                  class="margin-left-1"
                >
                  Elbert Huel
                </span>
              </div>
            </td>
            <td
              role="cell"
            >
              Business Owner, ISSO, API Contact, Budget Analyst, Support Staff
            </td>
            <td
              role="cell"
            >
              <div
                class="text-no-wrap"
              >
                <a
                  class="usa-link text-primary"
                  href="/systems/000-0000-1/team/edit/team-member?workspace"
                >
                  Edit roles
                </a>
                <button
                  class="usa-button usa-button--unstyled margin-left-1 text-error"
                  data-testid="button"
                  type="button"
                >
                  Remove
                </button>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              <div
                class="display-flex flex-align-center"
              >
                <div
                  class="display-flex flex-align-center flex-justify-center minw-4 circle-4 bg-green-5 "
                  data-testid="avatar--basic"
                >
                  VD
                </div>
                <span
                  class="margin-left-1"
                >
                  Vickie Denesik
                </span>
              </div>
            </td>
            <td
              role="cell"
            >
              Business Owner, Subject Matter Expert (SME)
            </td>
            <td
              role="cell"
            >
              <div
                class="text-no-wrap"
              >
                <a
                  class="usa-link text-primary"
                  href="/systems/000-0000-1/team/edit/team-member?workspace"
                >
                  Edit roles
                </a>
                <button
                  class="usa-button usa-button--unstyled margin-left-1 text-error"
                  data-testid="button"
                  type="button"
                >
                  Remove
                </button>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              <div
                class="display-flex flex-align-center"
              >
                <div
                  class="display-flex flex-align-center flex-justify-center minw-4 circle-4 bg-yellow-5 "
                  data-testid="avatar--basic"
                >
                  MS
                </div>
                <span
                  class="margin-left-1"
                >
                  Mariam Schaden
                </span>
              </div>
            </td>
            <td
              role="cell"
            >
              Project Lead, Business Question Contact
            </td>
            <td
              role="cell"
            >
              <div
                class="text-no-wrap"
              >
                <a
                  class="usa-link text-primary"
                  href="/systems/000-0000-1/team/edit/team-member?workspace"
                >
                  Edit roles
                </a>
                <button
                  class="usa-button usa-button--unstyled margin-left-1 text-error"
                  data-testid="button"
                  type="button"
                >
                  Remove
                </button>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              <div
                class="display-flex flex-align-center"
              >
                <div
                  class="display-flex flex-align-center flex-justify-center minw-4 circle-4 bg-blue-10 "
                  data-testid="avatar--basic"
                >
                  SB
                </div>
                <span
                  class="margin-left-1"
                >
                  Sasha Barrows
                </span>
              </div>
            </td>
            <td
              role="cell"
            >
              Project Lead, Contracting Officer's Representative (COR), Budget Analyst, System Maintainer, AI Contact
            </td>
            <td
              role="cell"
            >
              <div
                class="text-no-wrap"
              >
                <a
                  class="usa-link text-primary"
                  href="/systems/000-0000-1/team/edit/team-member?workspace"
                >
                  Edit roles
                </a>
                <button
                  class="usa-button usa-button--unstyled margin-left-1 text-error"
                  data-testid="button"
                  type="button"
                >
                  Remove
                </button>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              <div
                class="display-flex flex-align-center"
              >
                <div
                  class="display-flex flex-align-center flex-justify-center minw-4 circle-4 bg-green-5 "
                  data-testid="avatar--basic"
                >
                  EG
                </div>
                <span
                  class="margin-left-1"
                >
                  Eliezer Grant
                </span>
              </div>
            </td>
            <td
              role="cell"
            >
              Government Task Lead (GTL), Contracting Officer's Representative (COR)
            </td>
            <td
              role="cell"
            >
              <div
                class="text-no-wrap"
              >
                <a
                  class="usa-link text-primary"
                  href="/systems/000-0000-1/team/edit/team-member?workspace"
                >
                  Edit roles
                </a>
                <button
                  class="usa-button usa-button--unstyled margin-left-1 text-error"
                  data-testid="button"
                  type="button"
                >
                  Remove
                </button>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              <div
                class="display-flex flex-align-center"
              >
                <div
                  class="display-flex flex-align-center flex-justify-center minw-4 circle-4 bg-yellow-5 "
                  data-testid="avatar--basic"
                >
                  JS
                </div>
                <span
                  class="margin-left-1"
                >
                  Jefferey Schmidt
                </span>
              </div>
            </td>
            <td
              role="cell"
            >
              Contracting Officer's Representative (COR)
            </td>
            <td
              role="cell"
            >
              <div
                class="text-no-wrap"
              >
                <a
                  class="usa-link text-primary"
                  href="/systems/000-0000-1/team/edit/team-member?workspace"
                >
                  Edit roles
                </a>
                <button
                  class="usa-button usa-button--unstyled margin-left-1 text-error"
                  data-testid="button"
                  type="button"
                >
                  Remove
                </button>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              <div
                class="display-flex flex-align-center"
              >
                <div
                  class="display-flex flex-align-center flex-justify-center minw-4 circle-4 bg-green-5 "
                  data-testid="avatar--basic"
                >
                  KT
                </div>
                <span
                  class="margin-left-1"
                >
                  King Tromp
                </span>
              </div>
            </td>
            <td
              role="cell"
            >
              Contracting Officer's Representative (COR), Subject Matter Expert (SME), Support Staff
            </td>
            <td
              role="cell"
            >
              <div
                class="text-no-wrap"
              >
                <a
                  class="usa-link text-primary"
                  href="/systems/000-0000-1/team/edit/team-member?workspace"
                >
                  Edit roles
                </a>
                <button
                  class="usa-button usa-button--unstyled margin-left-1 text-error"
                  data-testid="button"
                  type="button"
                >
                  Remove
                </button>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              <div
                class="display-flex flex-align-center"
              >
                <div
                  class="display-flex flex-align-center flex-justify-center minw-4 circle-4 bg-red-cool-10 "
                  data-testid="avatar--basic"
                >
                  AW
                </div>
                <span
                  class="margin-left-1"
                >
                  Abner Waelchi
                </span>
              </div>
            </td>
            <td
              role="cell"
            >
              Technical System Issues Contact
            </td>
            <td
              role="cell"
            >
              <div
                class="text-no-wrap"
              >
                <a
                  class="usa-link text-primary"
                  href="/systems/000-0000-1/team/edit/team-member?workspace"
                >
                  Edit roles
                </a>
                <button
                  class="usa-button usa-button--unstyled margin-left-1 text-error"
                  data-testid="button"
                  type="button"
                >
                  Remove
                </button>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              <div
                class="display-flex flex-align-center"
              >
                <div
                  class="display-flex flex-align-center flex-justify-center minw-4 circle-4 bg-yellow-5 "
                  data-testid="avatar--basic"
                >
                  FB
                </div>
                <span
                  class="margin-left-1"
                >
                  Forest Brown
                </span>
              </div>
            </td>
            <td
              role="cell"
            >
              Data Center Contact
            </td>
            <td
              role="cell"
            >
              <div
                class="text-no-wrap"
              >
                <a
                  class="usa-link text-primary"
                  href="/systems/000-0000-1/team/edit/team-member?workspace"
                >
                  Edit roles
                </a>
                <button
                  class="usa-button usa-button--unstyled margin-left-1 text-error"
                  data-testid="button"
                  type="button"
                >
                  Remove
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="grid-row grid-gap grid-gap-lg"
      >
        <nav
          aria-label="Pagination"
          class="usa-pagination padding-bottom-1 desktop:grid-col-fill flex-justify-start margin-left-neg-1 desktop:padding-bottom-0 desktop:margin-bottom-0"
          data-testid="table-pagination"
        >
          <ul
            class="usa-pagination__list"
          >
            <li
              class="usa-pagination__item usa-pagination__page-no"
            >
              <div>
                <div
                  class="usa-pagination__item"
                >
                  <button
                    aria-label="Page 1"
                    class="usa-pagination__button usa-current"
                    type="button"
                  >
                    1
                  </button>
                </div>
              </div>
              <div>
                <div
                  class="usa-pagination__item"
                >
                  <button
                    aria-label="Page 2"
                    class="usa-pagination__button"
                    type="button"
                  >
                    2
                  </button>
                </div>
              </div>
            </li>
            <li
              class="usa-pagination__item usa-pagination__arrow"
            >
              <button
                aria-label="Next page"
                class="usa-pagination__link usa-pagination__previous-page"
                type="button"
              >
                <span
                  class="usa-pagination__link-text"
                >
                  Next &gt;
                </span>
              </button>
            </li>
          </ul>
        </nav>
        <div
          class="desktop:margin-top-2 desktop:grid-col-auto"
        >
          <select
            class="usa-select margin-top-0 width-auto"
            data-testid="table-page-size"
            id="table-page-size"
            name="tablePageSize"
          >
            <option
              value="5"
            >
              Show 5
            </option>
            <option
              value="10"
            >
              Show 10
            </option>
            <option
              value="25"
            >
              Show 25
            </option>
            <option
              value="50"
            >
              Show 50
            </option>
            <option
              value="100"
            >
              Show 100
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
